apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/PooledSession.cs

```diff
@@ -0,0 +1,319 @@
+            return RecordSuccessAndExpiredSessions(Client.RollbackAsync(request, settings));
+        }
+
+        /// <summary>
+        /// Executes a PartitionQuery RPC asynchronously.
+        /// </summary>
+        /// <param name="request">The partitioning request. Must not be null. The request will be modified with session details
+        /// from this object.</param>
+        /// <param name="timeoutSeconds">The timeout for this RPC, in seconds.</param>
+        /// <param name="cancellationToken">An optional token for canceling the call.</param>
+        /// <returns>A task representing the asynchronous operation. When the task completes, the result is the response from the RPC.</returns>
+        public Task<PartitionResponse> PartitionQueryAsync(PartitionQueryRequest request, int timeoutSeconds, CancellationToken cancellationToken)
+        {
+            CheckNotDisposed();
+            GaxPreconditions.CheckNotNull(request, nameof(request));
+            GaxPreconditions.CheckState(TransactionId != null, "Cannot call PartitionQueryAsync with no associated transaction");
+            request.SessionAsSessionName = SessionName;
+            request.Transaction = new TransactionSelector { Id = TransactionId };
+
+            var settings = CreateSettings(Client.Settings.PartitionQuerySettings, timeoutSeconds, cancellationToken);
+            return RecordSuccessAndExpiredSessions(Client.PartitionQueryAsync(request, settings));
+        }
+
+        /// <summary>
+        /// Creates a <see cref="ReliableStreamReader"/> for the given request
+        /// </summary>
+        /// <param name="request">The query request. Must not be null. The request will be modified with session and transaction details
+        /// from this object. If this object's <see cref="TransactionId"/> is null, the request's transaction is not modified.</param>
+        /// <param name="timeoutSeconds">The timeout for this RPC, in seconds.</param>
+        /// <returns>A <see cref="ReliableStreamReader"/> for the streaming SQL request.</returns>
+        public ReliableStreamReader ExecuteSqlStreamReader(ExecuteSqlRequest request, int timeoutSeconds)
+        {
+            CheckNotDisposed();
+            GaxPreconditions.CheckNotNull(request, nameof(request));
+            if (TransactionId != null)
+            {
+                request.Transaction = new TransactionSelector { Id = TransactionId };
+            }
+            return Client.GetSqlStreamReader(request, _session, timeoutSeconds);
+        }
+
+        /// <summary>
+        /// Executes an ExecuteSql RPC asynchronously.
+        /// </summary>
+        /// <param name="request">The query request. Must not be null. The request will be modified with session and transaction details
+        /// from this object. If this object's <see cref="TransactionId"/> is null, the request's transaction is not modified.</param>
```

Comment:
Yes and yes. In particular, this can have a transaction selector of a single-read operation, e.g. "staleness of at most 5 seconds".
Spanner transactions are complicated :(